DataDog/datadog-cdk-constructs
# adds support for singleton functions

### What does this PR do?

Adds support for Singleton functions.

### Motivation

https://datadoghq.atlassian.net/browse/SVLS-3474

### Testing Guidelines

* Added unit test to verify `addLambdaFunctions` handles Singleton functions as expected
* Added integration test for adding Datadog extension and layer to Singleton function
  * `./v2/scripts/run_integration_tests.sh` (with `UPDATE_SNAPSHOTS-true` to generate snapshot for singleton function)
* Tested manually in sandbox
  * `npx cdk --app lib/sample/index.js deploy`

### Additional Notes

The Lambda function in the SingletonFunction class is in the `lambdaFunction` property so before making any updates to Singleton functions the Lambda function is extracted from this `lambdaFunction` property.

* Updated `test-utils.findDatadogSubscriptionFilters` to properly handle Singleton functions in unit test
* Added singleton function handling to `datadog.addGitCommitMetadata` since it is also a public method along with `datadog.addLambdaFunctions`

### Types of Changes


### Check all that apply
